Add GooseAI, docs, and tests

conceptofmind opened the pull request:
Add GooseAI integration, Documentation, and tests. Hopefully, I am not missing anything.

Usage:
```python
import os
from langchain.llms import GooseAI
from langchain import PromptTemplate, LLMChain

os.environ["GOOSEAI_API_KEY"] = ""

llm = GooseAI(max_tokens=10)

template = """Question: {question}

Answer: Let's think step by step."""

prompt = PromptTemplate(template=template, input_variables=["question"])

llm_chain = LLMChain(prompt=prompt, llm=llm)

question = "What NFL team won the Super Bowl in the year Justin Beiber was born?"

print(llm_chain.run(question))
```

conceptofmind committed:
Add GooseAI, docs, and tests

conceptofmind committed:
fix test

conceptofmind commented:
Had to fix a test.

conceptofmind commented:
There might be one slight issue. I will have to check tomorrow.

hwchase17 committed:
chat vector db chain (#902)

hwchase17 committed:
Harrison/unstructured support (#903)

hwchase17 committed:
update docs (#905)

hwchase17 committed:
bump version to 0078 (#908)

JohnNay commented on langchain/llms/gooseai.py:
```diff
@@ -0,0 +1,134 @@
+
+logger = logging.getLogger(__name__)
+
+class GooseAI(LLM, BaseModel):
```
If a lot of this is very similar to the OpenAI class, should we abstract that similarity and then create this class from that abstraction?

conceptofmind replied:
I was thinking about just adding an abstraction similar to AzureOpenAI but I was concerned that if GooseAI or OpenAI decide to change their APIs in some sort of manner it could cause conflicts. I did not know if it would be easier to diagnose or maintain if they are separated. 

I am completely open to just adding a similar class GooseAI class abstraction if you think that is better. Whatever works best!

jamescalam committed:
added i_end in batch extraction (#907)

Fix for issue #906 

Switches `[i : i + batch_size]` to `[i : i_end]` in Pinecone
`from_texts` method

hwchase17 committed:
Optionally return shell output on incorrect command (#894) (#899)

This allows the LLM to correct its previous command by looking at the
error message output to the shell.

Additionally, this uses subprocess.run because that is now recommended
over subprocess.check_output:

https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module

Co-authored-by: Amos Ng <me@amos.ng>

kwhuo68 committed:
Add pinecone integration test (#911)

Basic integration test for pinecone

hwchase17 committed:
add unstructured examples (#913)

hwchase17 committed:
Harrison/inference endpoint (#861)

Co-authored-by: Eno Reyes <enoreyes@gmail.com>

hwchase17 committed:
Harrison/sql rows (#915)

Co-authored-by: Jon Luo <20971593+jzluo@users.noreply.github.com>

hwchase17 committed:
Harrison/prompt template prefix (#888)

Co-authored-by: Gabriel Simmons <simmons.gabe@gmail.com>

hwchase17 committed:
analyze document (#731)

add analyze document chain, which does text splitting and then analysis

hwchase17 committed:
fix docs custom prompt template (#917)

hwchase17 committed:
add reqs (#918)

hwchase17 committed:
remove unused buffer (#919)

agola11 committed:
add GoogleDriveLoader (#914)

only deal with docs files for now

hwchase17 committed:
Harrison/save faiss (#916)

Co-authored-by: Shrey Joshi <shreyjoshi2004@gmail.com>

hwchase17 committed:
Harrison/obsidian (#920)

hwchase17 committed:
Harrison/athropic (#921)

Co-authored-by: Mike Lambert <mlambert@gmail.com>
Co-authored-by: mrbean <sam@you.com>
Co-authored-by: mrbean <43734688+sam-h-bean@users.noreply.github.com>
Co-authored-by: Ivan Vendrov <ivendrov@gmail.com>

hwchase17 committed:
Harrison/ver0079 (#927)

VinceDerPrince committed:
Fix typos (#929)

accomplisehd -> accomplished

agola11 committed:
Add asyncio support for LLM (OpenAI), Chain (LLMChain, LLMMathChain), and Agent (#841)

Supporting asyncio in langchain primitives allows for users to run them
concurrently and creates more seamless integration with
asyncio-supported frameworks (FastAPI, etc.)

Summary of changes:

**LLM**
* Add `agenerate` and `_agenerate`
* Implement in OpenAI by leveraging `client.Completions.acreate`

**Chain**
* Add `arun`, `acall`, `_acall`
* Implement them in `LLMChain` and `LLMMathChain` for now

**Agent**
* Refactor and leverage async chain and llm methods
* Add ability for `Tools` to contain async coroutine
* Implement async SerpaPI `arun`

Create demo notebook.

Open questions:
* Should all the async stuff go in separate classes? I've seen both
patterns (keeping the same class and having async and sync methods vs.
having class separation)

hwchase17 committed:
Harrison/add roam loader (#939)

hwchase17 committed:
bump version to 0080 (#941)

eltociear committed:
Update huggingface_hub.ipynb (#944)

HuggingFace -> Hugging Face

jefftangx committed:
docs: fix spelling typo (#934)

hwchase17 committed:
gutenberg books (#946)

Co-authored-by: zanderchase <zander@unfold.ag>
Co-authored-by: Harrison Chase <harrisonchase@Harrisons-MBP.attlocal.net>

concaption committed:
Update readthedocs.py (#943)

Sometimes, the docs may be empty. For example for the text =
soup.find_all("main", {"id": "main-content"}) was an empty list. To
cater to these edge cases, the clean function needs to be checked if it
is empty or not.

hwchase17 committed:
add docs for steamship deployment (#949)

Co-authored-by: Harrison Chase <harrisonchase@Harrisons-MBP.attlocal.net>

conceptofmind committed:
Merge branch 'master' into master

conceptofmind commented:
Resolve conflict with newly added import in update

zanderchase committed:
adding webpage loading logic (#942)

hwchase17 committed:
Harrison/youtube fixes (#955)

Co-authored-by: Ji <jizhang.work@gmail.com>
Co-authored-by: Harrison Chase <harrisonchase@Harrisons-MBP.attlocal.net>

hwchase17 committed:
bump version 0081 (#956)

Co-authored-by: Harrison Chase <harrisonchase@Harrisons-MBP.attlocal.net>

conceptofmind committed:
Merge branch 'hwchase17:master' into master

conceptofmind committed:
Fix typing

conceptofmind committed:
Fix typing

conceptofmind committed:
Merge branch 'master' of https://github.com/conceptofmind/langchain

conceptofmind committed:
black formatting

conceptofmind commented:
Closing this until I can figure out formatting errors. Going to open one that handles issues with linting and adds support for all LLM providers.
